pkgs/applications/networking/geph/default.nix

```diff
@@ -0,0 +1,127 @@
+{ lib
+, stdenvNoCC
+, rustPlatform
+, fetchFromGitHub
+, buildNpmPackage
+, perl
+, pkg-config
+, glib
+, webkitgtk
+, libappindicator-gtk3
+, libayatana-appindicator
+, cairo
+, openssl
+}:
+
+let
+  geph-meta = with lib; {
+    description = "A modular Internet censorship circumvention system designed specifically to deal with national filtering.";
+    homepage = "https://geph.io";
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ penalty1083 ];
+  };
+in
+{
+  cli = rustPlatform.buildRustPackage rec {
+    pname = "geph4-client";
+    version = "4.7.0-beta.4";
+
+    src = fetchFromGitHub {
+      owner = "geph-official";
+      repo = pname;
+      rev = "v${version}";
+      hash = "sha256-elPZqkrh++1hoYpmhDJlBLmr0ELVUESkRAj+mu8UkuU=";
+    };
+
+    cargoHash = "sha256-D4o9i+vjYgIcG3YslAck19uJsglsGaO6MLXdmsTSnVw=";
+
+    nativeBuildInputs = [ perl ];
+
+    meta = geph-meta // {
+      license = with lib.licenses; [ gpl3Only ];
+    };
+  };
+
+  gui = stdenvNoCC.mkDerivation rec {
+    pname = "geph-gui";
+    version = "4.7.0-beta.4";
+
+    src = fetchFromGitHub {
+      owner = "geph-official";
+      repo = "gephgui-pkg";
+      rev = "6439945d6baa4b2a43bed7fa5a55ddbfcfc099be";
+      hash = "sha256-0cisnOJFaZHMb+KXdV5pbAayegtDPUHSWG/NMAuQJ4A=";
+      fetchSubmodules = true;
```

Comment:
Don't we want o use our gephgui-wry instead of https://github.com/geph-official/gephgui-pkg/blob/master/.gitmodules#L1-L6 ?

Reply:
The submodule is only used to keep them in sync. Our gephgui-wry is used.